shapely/strtree.py

```diff
@@ -84,8 +84,9 @@ def __init__(self, geoms):
     def _init_tree_handle(self, geoms):
         # GEOS STRtree capacity has to be > 1
         self._tree_handle = lgeos.GEOSSTRtree_create(max(2, len(geoms)))
-        for geom in geoms:
-            lgeos.GEOSSTRtree_insert(self._tree_handle, geom._geom, ctypes.py_object(geom))
+        self._idxs = list(range(len(geoms)))
+        for idx, geom in zip(self._idxs, geoms):
```

Comment:
I should add a comment about it to clarify, but we need to store `self._idxs` for memory management purposes (python needs to keep track of those integers, to avoid garbage collecting them while they are stored in the C++ GEOS STRtree). 

Still, this `for` loop could be simplified to use `enumerate`, but since we have the `self._idxs` anyway, it seems a bit more explicit to use it here.

Reply:
Definitely worthy of a comment making that clear. If we need to hold on to the indexes then the way you have it makes total sense.